tck/features/MergeRelationshipAcceptance.feature

```diff
@@ -0,0 +1,833 @@
+      """
+      MERGE (a {id: 2})-[r:KNOWS]-(b {id: 1})
+      RETURN startNode(r).id AS s, endNode(r).id AS e
+      """
+    Then the result should be:
+      | s | e |
+      | 2 | 1 |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +properties    | 2 |
+
+  Scenario: Match outgoing relationship when direction unspecified
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {id: 1}), (b {id: 2})
+      CREATE (a)-[:KNOWS]->(b)
+      """
+    When executing query:
+      """
+      MERGE (a {id: 2})-[r:KNOWS]-(b {id: 1})
+      RETURN r
+      """
+    Then the result should be:
+      | r        |
+      | [:KNOWS] |
+    And no side effects
+
+  Scenario: Match both incoming and outgoing relationships when direction unspecified
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {id: 2}), (b {id: 1}), (c {id: 1}), (d {id: 2})
+      CREATE (a)-[:KNOWS {name: 'ab'}]->(b)
+      CREATE (c)-[:KNOWS {name: 'cd'}]->(d)
+      """
+    When executing query:
+      """
+      MERGE (a {id: 2})-[r:KNOWS]-(b {id: 1})
+      RETURN r
+      """
+    Then the result should be:
+      | r                     |
+      | [:KNOWS {name: 'ab'}] |
+      | [:KNOWS {name: 'cd'}] |
+    And no side effects
+
+  Scenario: Fail when imposing new predicates on already bound variable
+    Given any graph
+    When executing query:
+      """
+      MERGE (a:Foo)-[r:KNOWS]->(a:Bar)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Using list properties via variable
+    Given an empty graph
+    When executing query:
+      """
+      CREATE (a:Foo), (b:Bar)
+      WITH a, b
+      UNWIND ["a,b", "a,b"] AS str
+      WITH a, b, split(str, ",") AS roles
+      MERGE (a)-[r:FB {foobar: roles}]->(b)
+      RETURN count(*)
+      """
+    Then the result should be:
+      | count(*) |
+      | 2        |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +labels        | 2 |
+      | +properties    | 1 |
+
+  Scenario: Matching using list property
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      CREATE (a)-[:T {prop: [42, 43]}]->(b)
+      """
+    When executing query:
+      """
+      MATCH (a:A), (b:B)
+      MERGE (a)-[r:T {prop: [42, 43]}]->(b)
+      RETURN count(*)
+      """
+    Then the result should be:
+      | count(*) |
+      | 1        |
+    And no side effects
+
+  Scenario: Using bound variables from other updating clause
+    Given an empty graph
+    When executing query:
+      """
+      CREATE (a)
+      MERGE (a)-[:X]->()
+      RETURN count(a)
+      """
+    Then the result should be:
+      | count(a) |
+      | 1        |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+
+  Scenario: `UNWIND` with multiple merges
```

Comment:
They are not needed. I can't recall why I started using them.

They are necessary when using parentheses, for things like functions.